net/gce.sh

```diff
@@ -501,25 +470,37 @@ EOF
     bootstrapLeaderAddress=$customAddress
   fi
 
-  cloud_Initialize "$prefix"
+  for zone in "${zones[@]}"; do
+    cloud_Initialize "$prefix" "$zone"
+  done
 
   cloud_CreateInstances "$prefix" "$prefix-bootstrap-leader" 1 \
-    "$imageName" "$bootstrapLeaderMachineType" "$fullNodeBootDiskSizeInGb" \
+    "$enableGpu" "$bootstrapLeaderMachineType" "${zones[0]}" "$fullNodeBootDiskSizeInGb" \
     "$startupScript" "$bootstrapLeaderAddress" "$bootDiskType"
 
-  cloud_CreateInstances "$prefix" "$prefix-fullnode" "$additionalFullNodeCount" \
-    "$imageName" "$fullNodeMachineType" "$fullNodeBootDiskSizeInGb" \
-    "$startupScript" "" "$bootDiskType"
+  num_zones=${#zones[@]}
+  numNodesPerZone=$((additionalFullNodeCount / num_zones))
+  numLeftOverNodes=$((additionalFullNodeCount % num_zones))
+  count=0
+  for zone in "${zones[@]}"; do
```

Comment:
bash has first-class `for ((i=0; i < ${#zones[@]}; i++ )) `

and then you can ${zones[i]} and skip `((count++))``